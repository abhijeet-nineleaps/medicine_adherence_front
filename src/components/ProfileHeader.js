import {GoogleSignin} from '@react-native-google-signin/google-signin';
import {useFocusEffect} from '@react-navigation/native';
import React from 'react';
import {View, Text, Image} from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import styles from './componentStyles/styles';

const ProfileHeader = () => {
  const [umg, imgstate] = React.useState('https://i.stack.imgur.com/l60Hf.png');
  const [name, namestate] = React.useState('');
  useFocusEffect(/* istanbul ignore next */() => {
    async function getuser() {
      try {
        if (!(await GoogleSignin.isSignedIn())) {
          imgstate('https://i.stack.imgur.com/l60Hf.png');
          await AsyncStorage.getItem('user_id');
          let username = await AsyncStorage.getItem('user_name');
          if (username !== null) {
            username = '';
          }
          namestate(username);
          return;
        }
        const user = await GoogleSignin.getCurrentUser();
        imgstate(user.user.photo);
        namestate(user.user.name);
      } catch (err) {}
    }
    getuser();
  });

  return (
    <View
      style={styles.container}>
      <Image
        source={{
          uri: umg,
        }}
        style={styles.img}
      />
      <View style={styles.nameConatiner}>
        <Text
          style={styles.nameText}>
          {name}
        </Text>
        <Text style={styles.text}>
          {'View and edit profile'}
        </Text>
      </View>
    </View>
  );
};

export default ProfileHeader;
