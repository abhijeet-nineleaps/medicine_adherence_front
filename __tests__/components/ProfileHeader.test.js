import React from "react";
import renderer from 'react-test-renderer';
import enableHooks from "jest-react-hooks-shallow";
import ProfileHeader from "../../src/components/ProfileHeader";
enableHooks(jest);
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));jest.mock("@react-native-google-signin/google-signin", () => ({
    default: jest.fn(),
  }));
  jest.mock("@react-navigation/native", () => ({
    ...jest.requireActual("@react-navigation/native"),
    useFocusEffect: jest.fn().mockImplementation((getuser) => getuser()),
  }));
jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useFocusEffect: jest.fn(),
}));
describe('Click send image', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(<ProfileHeader/>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
});