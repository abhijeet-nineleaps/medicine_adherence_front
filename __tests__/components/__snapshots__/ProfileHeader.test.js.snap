// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click send image renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3743ab",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": 20,
      "padding": 20,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://i.stack.imgur.com/l60Hf.png",
      }
    }
    style={
      Object {
        "borderRadius": 70,
        "height": 100,
        "marginBottom": 9,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 15,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      View and edit profile
    </Text>
  </View>
</View>
`;
